internal/databases/greenplum/backup_fetch_handler_test.go

```diff
@@ -50,3 +51,131 @@ func TestPrepareContentIDsToFetch(t *testing.T) {
 		assert.Equal(t, tc.contentIDsToFetch, contentIDsToFetch)
 	}
 }
+
+func TestBuildFetchCommand(t *testing.T) {
+	beforeValue := config.CfgFile
+	defer func() {
+		config.CfgFile = beforeValue
+	}()
+	config.CfgFile = "testConfig"
+
+	testcases := []struct {
+		handler   *FetchHandler
+		contentID int
+		cmdLine   string
+	}{
+		{
+			&FetchHandler{
```

Comment:
I'm not sure if I should use `NewFetchHandler` or construct an object like this. At first I tried to use `NewFetchHandler` but it quicky became very complicated.

But I guess I should use `NewFetchHandler`. It's easer to refactor. I think no one will be happy if they change `cluster.SeqConfig` object and this test will fail. 

But I need some tips how to call `NewFetchHandler` correctly without a headache.

Reply:
Well, if you want to use `NewFetchHandler `, the most difficult things seem to be `BackupSentinelDto` and `SegConfigMaker`. From the first one you need only segments, so you can define it like
```
dto := BackupSentinelDto{
		Segments: []SegmentMetadata{
			{
			DatabaseID: 1,
			ContentID: 1,
			Role: Primary,
			Port: 1234,
			Hostname: "name",
			DataDir: "datadir",		
			BackupID: "backup",
			},
		},
	}
```
And `SegConfigMaker` you can make from string:
```
seg, err := NewRestoreCfgSegMaker(strings.NewReader(`{"segments":{"1":{"hostname":"name", "port":1234, "data_dir": "dir"}}}`))
```

Reply:
@debebantur , thank you

> if you want to use

Do you think I should do it the way you suggested or it's okey to leave it like this?

And with machinery all this variables is it possible to declare it in `testcases` array?

Reply:
I believe that it is ok to leave it, anyway we need to test only `buildFetchCommand`. Also it seems difficult to declare all this structures in testcases array.